pkg/rancher-desktop/utils/shortcuts.ts

```diff
@@ -0,0 +1,207 @@
+import os from 'os';
+
+import { BrowserWindow } from 'electron';
+
+import { toArray } from '@pkg/utils/array';
+
+type Platform = 'darwin' | 'linux' | 'win32';
+
+const Shortcut = {
+  /**
+   * TODO implement shift, alt
+   */
+  key: (shortcut: Shortcut | Electron.Input) => {
+    return `${ shortcut.meta ? 'Cmd+' : '' }${ shortcut.control ? 'Ctrl+' : '' }${ shortcut.key }`;
+  },
+};
+
+export interface Shortcut {
+  platform?: Platform | Platform[];
+  meta?: boolean;
+  shift?: boolean;
+  control?: boolean;
+  alt?: boolean;
+  key: string;
+}
+
+interface ShortcutExt extends Shortcut {
+  callback: () => void;
+}
+
+function matchPlatform(shortcut: Shortcut): boolean {
+  if (shortcut.platform) {
+    return toArray(shortcut.platform).includes(os.platform());
+  }
+
+  return true;
+}
+
+function currentPlatform(shortcuts: Shortcut[], action: (s: Shortcut) => void) {
```

Comment:
I'm wondering if there's a better name for this function other than `currentPlatform`. It wasn't immediately clear to me what this was supposed to be doing when I was reviewing its usage. 

I can see that we are trying to apply some action (e.g. Add/Remove) for each shortcut on the current platform. I don't know if there's something we can do with that.. Bare minimum, I think we should document this function so that its purpose is clear.

Reply:
I think we can remove `currentPlatform`, leaving a loop for each shortcut is quite self-explanatory:

```
shortcuts.forEach((s) => {
  if (matchPlatform(s)) {
    this.windows[id].removeShortcut(s);
  }
});
```

Reply:
I agree with this approach. It's definitely easier for me to understand at a glance 😁